src/dreyfus_index.erl

```diff
@@ -157,9 +159,28 @@ handle_call(info, _From, State) -> % obsolete
     Reply = info_int(State#state.index_pid),
     {reply, Reply, State}.
 
+handle_cast({ddoc_updated, DDocResult}, State) ->
+    #state{index=#index{sig=CurrentSig}, waiting_list = WaitList} = State,
+    case check_if_index_is_deleted(DDocResult, CurrentSig) of
+        true ->
```

Comment:
this case statement will crash when the index isn't deleted as you haven't written the `false` case. I would also suggest collapsing this to one case statement using `{check_if_index_is_deleted(DDocResult, CurrentSig), WaitList}`